catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/data/impl/MetacardImpl.java

```diff
@@ -757,9 +757,9 @@ public void setAttribute(Attribute attribute) {
       wrappedMetacard.setAttribute(attribute);
     } else {
       String name = attribute.getName();
-      Serializable value = attribute.getValue();
+      List<Serializable> values = attribute.getValues()
```

Comment:
I don't think this will work because a common way to clear a metacard value is to do a  setAttribute(<att-name>,null). The getValues() will return a List with a null value in it so it will just add it to the map instead of removing it.

Reply:
Are you saying I should update the check to be 
```
if (values != null && values.size() > 0) ...add to map
```

Reply:
The getValues() mentioned above will return a list of size 1 with the first element being null so you would need to check if the size is 1 and the first element is null then remove it from the map which might defeat the point of this change.

Reply:
This is to fix the case where the list has size > 1, but the first item is null. If the list only has 1 null item in it then that can be removed.